presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -683,6 +685,36 @@ protected Scope visitQuery(Query node, Optional<Scope> scope)
                     .withParent(withScope)
                     .withRelationType(RelationId.of(node), queryBodyScope.getRelationType())
                     .build();
+
+            // create a multimap of tables in a query to columns used for each table
+            Multimap<QualifiedObjectName, String> tableColumnMap = analysis.getColumnReferenceFields().values().stream()
+                    .filter(fieldId -> {
+                        // first filter out column references that don't directly reference tables
+                        Node sourceNode = fieldId.getRelationId().getSourceNode();
+                        return sourceNode instanceof Table
+                                && analysis.getScope(sourceNode)
```

Comment:
Why do you need this?

Reply:
for queries like `WITH a AS (SELECT 1) (SELECT * FROM a);`  you end up with a `sourceNode` of type `TABLE` for the `SELECT * FROM a`, but it doesn't have an origin table.